swarm/pss/snapshot_test.go

```diff
@@ -0,0 +1,433 @@
+		t.Fatal("cannot specify only one of flags --nodes and --messages")
+	} else if *runNodes > 0 {
+		t.Run(fmt.Sprintf("%d/%d", *runMessages, *runNodes), testProxNetwork)
+	} else {
+		t.Run("1/4", testProxNetwork)
+	}
+}
+
+// This tests generates a sequenced number of messages with random addresses.
+// It then calculates which nodes in the network have the address of each message
+// within their nearest neighborhood depth, and stores them as recipients.
+// Upon sending the messages, it verifies that the respective message is passed to the message handlers of these recipients.
+// It will fail if a recipient handles a message it should not, or if after propagation not all expected messages are handled (timeout)
+func testProxNetwork(t *testing.T) {
+	resetTestVariables()
+	msgCount, nodeCount := getCmdParams(t)
+	handlerContextFuncs := make(map[Topic]handlerContextFunc)
+	handlerContextFuncs[topic] = nodeMsgHandler
+	services := newProxServices(true, handlerContextFuncs, kademlias)
+	snap := readSnapshot(t, nodeCount)
+	sim = simulation.New(services)
+	defer sim.Close()
+	err := sim.Net.Load(&snap)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	defer cancel()
+	err = sim.WaitTillSnapshotRecreated(ctx, snap)
+	if err != nil {
+		t.Fatalf("failed to recreate snapshot: %s", err)
+	}
+	assingTestVariables(sim, msgCount)
+	result := sim.Run(ctx, runFunc)
+	if result.Error != nil {
+		// context deadline exceeded
+		// however, it might just mean that not all possible messages are received
+		// now we must check if all required messages are received
+		log.Debug("--------------------------------------------------------------------------------", "rcv", msgCnt)
+		if msgCnt < msgsToReceive {
+			t.Fatal(result.Error)
+		}
+	}
+	t.Logf("completed %d", result.Duration)
+}
+
+func sendAllMsgs(sim *simulation.Simulation, msgs [][]byte, senders map[int]enode.ID) {
+	for i, msg := range msgs {
+		log.Debug("sending msg", "idx", i, "from", senders[i])
+		nodeClient, err := sim.Net.GetNode(senders[i]).Client()
+		if err != nil {
+			log.Crit(err.Error())
+		}
+		var uvarByte [8]byte
+		binary.PutUvarint(uvarByte[:], uint64(i))
+		nodeClient.Call(nil, "pss_sendRaw", hexutil.Encode(msg), hexutil.Encode(topic[:]), hexutil.Encode(uvarByte[:]))
+	}
+	log.Debug("all messages sent")
+}
+
+func runFunc(ctx context.Context, sim *simulation.Simulation) error {
+	go handlerChannelListener(ctx)
+	go sendAllMsgs(sim, msgs, senders)
+	received := 0
+
+	// collect incoming messages and terminate with corresponding status when message handler listener ends
+	for {
+		select {
+		case err := <-errC:
+			return err
+		case hn := <-msgC:
+			received++
+			log.Debug("msg received", "msgs_received", received, "total_expected", msgsToReceive, "id", hn.id, "serial", hn.serial)
+			if received >= maxMessages {
+				close(doneC)
+				return nil
+			}
+		}
+	}
+	return nil
```

Comment:
unreacable code?

Reply:
some compilers might not detect this, and they will complain that function does not return value.